app/services/order_factory.rb

```diff
@@ -56,17 +57,24 @@ def build_line_items
   end
 
   def build_item_from(attrs)
-    @order.line_items.build(
-      attrs.merge(skip_stock_check: opts[:skip_stock_check])
-    )
+    line_item = @order.line_items.build
+    line_item.variant_id = attrs[:variant_id]
+    line_item.quantity = attrs[:quantity]
+    line_item.skip_stock_check = opts[:skip_stock_check]
   end
 
   def set_user
     @order.update_attribute(:user_id, customer.user_id)
   end
 
   def set_addresses
+    byebug
+    # this mass assignment isnt working... we need an alternative...
     @order.update_attributes(attrs.slice(:bill_address_attributes, :ship_address_attributes))
```

Comment:
We're assigning nested attributes for an associated object here, right? 

I think it's a bit different in Rails 4: https://stackoverflow.com/a/23495802

Reply:
do you want to tackle this one @Matt-Yorkley? I havent really started, this is all I have so far and I still dont know how to adapt to rails 4.

Reply:
this is #4820